plugins/helpers.nix

```diff
@@ -68,23 +119,71 @@ rec {
     name,
     description,
     extraPlugins ? [],
+    extraConfigLua ? "",
+    extraConfigVim ? "",
     options ? {},
     ...
   }: let
     cfg = config.programs.nixvim.plugins.${name};
     # TODO support nested options!
-    pluginOptions = mapAttrs (k: v: v.option) options;
+    moduleOptions = (mapAttrs (k: v: v.option) options);
+    # // {
+      # extraConfig = mkOption {
+      #   type = types.attrs;
+      #   default = {};
+      #   description = "Place any extra config here as an attibute-set";
+      # };
+    # };
+
     globals = mapAttrs' (name: opt: {
       name = opt.global;
       value = if cfg.${name} != null then opt.value cfg.${name} else null;
     }) options;
   in {
     options.programs.nixvim.plugins.${name} = {
       enable = mkEnableOption description;
-    } // pluginOptions;
+    } // moduleOptions;
+
+    config.programs.nixvim = mkIf cfg.enable {
+      inherit extraPlugins extraConfigVim globals;
+      extraConfigLua =
+        if stringLength extraConfigLua > 0 then
+          "do -- config scope: ${name}\n" + extraConfigLua + "\nend"
+        else "";
+    };
+  };
+
+  mkLuaPlugin = {
```

Comment:
Only thing I would add here is the keybinds option.

Reply:
I am not sure what you mean. Do you mean the keybinding options from barbar.nix?

Reply:
Yeah, I think this would be a great place to implement that. Can be merged as-is for now though, and I'll work on that later! Another nice option would be `extraOptions`, so that a user could add options to the setup function (as specified in #30)

Reply:
Did you miss the `extraConfigLua` option in line 149? Otherwise, i am not sure what you mean

Reply:
I meant that the auto-generated plugin code could add an extraOptions option that lets a user add their own options to the setup table

Reply:
Sorry, I missed the line, I meant line 174.
This adds `extraConfig` to the options, just like the `enable` option.